interfaces/apparmor/template_vars.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package apparmor
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/interfaces/dbus"
+	"github.com/ubuntu-core/snappy/snap"
+)
+
+// legacyVariablees returns text defining some apparmor variables that work
+// with legacy apparmor templates.
+//
+// The variables are expanded by apparmor parser. They are (currently):
+//  - APP_APPNAME
+//  - APP_ID_DBUS
+//  - APP_PKGNAME_DBUS
+//  - APP_PKGNAME
+//  - APP_VERSION
+//  - INSTALL_DIR
+// They can be changed but this has to match changes in template.go.
+//
+// In addition, the set of variables listed here interacts with old-security
+// interface since there the base template is provided by a particular 3rd
+// party snap, not by snappy.
+func legacyVariables(appInfo *snap.AppInfo) string {
+	var buf bytes.Buffer
+	fmt.Fprintf(&buf, "@{APP_APPNAME}=\"%s\"\n", appInfo.Name)
+	// TODO: replace with app.SecurityTag()
+	fmt.Fprintf(&buf, "@{APP_ID_DBUS}=\"%s\"\n",
+		dbus.SafePath(fmt.Sprintf("%s.%s_%s_%s",
+			appInfo.Snap.Name, appInfo.Snap.Developer, appInfo.Name, appInfo.Snap.Version)))
+	// XXX: How is this different from APP_ID_DBUS?
```

Comment:
APP_ID was previously defined as '<snap>_<appname>_<version>'. So the apparmor variables were:
APP_PKGNAME=<snap>
APP_APPNAME=<appname>
APP_VERSION=<version>

In this manner, APP_ID_DBUS was '<snap>_<appname>_<version>' translated to a dbus string, and APP_PKGNAME_DBUS was '<snap>' translated to a dbus string.